File: 序号/436.寻找右区间.kt
fun findRightInterval(intervals: Array<IntArray>): IntArray {
  // 右区间为 一个区间的start >= 另一区间的end，返回所有区间的右区间的索引
  // 用treemap记录所有区间的start和索引，遍历区间拿到大于右区间的值，从map中找到最近的左区间
  val dict = sortedMapOf<Int,Int>()
  intervals.forEachIndexed { index, ints ->
    dict[ints[0]]=index
  }
  val ans = IntArray(intervals.size){-1}
  intervals.forEachIndexed { index, ints ->
    for ((key,value) in dict){
      if (key>=ints[1]){
        ans[index]=value
        break
      }
    }
  }
  return ans
}

File: 序号/436. 寻找右区间.py
class Solution:
    def findRightInterval(self, intervals: List[List[int]]) -> List[int]:
        for idx,interval in enumerate(intervals):
            interval.append(idx)
        intervals.sort()
        n=len(intervals)
        ans=[-1]*n
        for _,end,idx in intervals:
            i = bisect.bisect_left(intervals,[end])
            if i<n:
                ans[idx]=intervals[i][2]
        return ans